File: Stack implementation/stack_using_static_array.c
/* Implementation of STACK using static array */
#include<stdio.h>
#include<stdlib.h>
#define CAPACITY 5

int stack[CAPACITY], top = -1;

// Defining function prototypes
int isEmpty(void);    // This function checks whether stack is empty of not.
int isFull(void);     // This function checks whether stack is full of not.
void push(int);       // This function is used to push an element in stack.
int pop(void);        // This function deletes the top most element of stack and returns it.
void peek(void);      // This function is used to see the topmost element of stack.
void traverse(void);  // This function is used to see all the elements of stack.


void main()
{
  int choice, item;

  while(1)
  {
    printf("Stack operations: \n");
    printf("1. Push\n");
    printf("2. Pop\n");
    printf("3. Peek\n");
    printf("4. Traverse\n");
    printf("5. Quit\n");

    printf("Enter your choice(1-5): ");
    scanf("%d", &choice);
    printf("\n");

    switch(choice)
    {
      case 1:
          printf("Enter element to push: ");
          scanf("%d", &item);
          push(item);
          break;
      case 2:
          item = pop();
          if(item != -999)
              printf("%d popped.\n\n", item);
          break;
      case 3:
          peek();
          break;
      case 4:
          traverse();
          break;
      case 5:
          exit(0);
      default:
          printf("Invalid input");
    }
  }
}

int isEmpty()
{
  if(top == -1)
      return 1;
  else
      return 0;
}

int isFull()
{
  if(top == CAPACITY - 1)
      return 1;
  else
      return 0;
}

void push(int ele)
{
  if(isFull())
    printf("Stack overflow.\n\n");
  else
  {
    top++;
    stack[top] = ele;
    printf("%d pushed\n\n", ele);
  }
}

int pop()
{
  if(isEmpty())
  {
    printf("Stack is empty.\n\n");
    return -999;
  }
  else
    return stack[top--];
}

void peek()
{
  if(isEmpty())
      printf("Stack is empty.\n\n");
  else
      printf("Peak element: %d\n\n", stack[top]);
}

void traverse()
{
  if(isEmpty())
      printf("Stack is empty.\n\n");
  else
  {
    int i;
    printf("Elements of stack from top to bottom:\n");
    for(i = top; i > -1; i--)
        printf("%d\n", stack[i]);
    printf("\n");
  }
}

File: Stack implementation/stack_using_static_array.java
// Implementation of stack using static array in java
import java.util.Scanner;
class Stack
{
  static Scanner in = new Scanner(System.in);
  static int stack[], size, top = -1;

  static
  {
    Stack.create_stack();
  }

  public static void main(String args[])
  {
    int choice, item;
    while(true)
    {
      System.out.println("\nStack operations:");
      System.out.println("1. Push");
      System.out.println("2. Pop");
      System.out.println("3. Peek");
      System.out.println("4. Traverse");
      System.out.println("5. Exit");

      System.out.print("Enter your choice: ");
      choice = in.nextInt();

      switch(choice)
      {
        case 1: System.out.print("Enter number to push: ");
                item = in.nextInt();
                Stack.push(item);
                break;

        case 2: item = Stack.pop();
                if(item == -999)
                  System.out.println("Stack is empty.");
                else
                  System.out.println(item + " popped.");
                break;

        case 3: item = Stack.peek();
                if(item == -999)
                  System.out.println("Stack is empty.");
                else
                  System.out.println("Peak item: " + item);
                break;

        case 4: Stack.traverse();
                break;

        case 5: System.exit(0);

        default:  System.out.println("Invalid input.");
      }
    }
  }

  public static void create_stack()
  {
    System.out.print("Enter size of stack: ");
    Stack.size = in.nextInt();
    Stack.stack = new int[Stack.size];
    System.out.println("Stack created with size: " + Stack.size);
  }

  public static boolean isFull()
  {
    if(top == size - 1)
      return true;          // If stack is full
    else
      return false;         // If stack is not full
  }

  public static boolean isEmpty()
  {
    if(top == -1)
      return true;          // If stack is empty
    else
      return false;         // If stack is not empty
  }

  public static void push(int element)
  {
    if(isFull())
      System.out.println("Stack is overflow.");
    else
    {
      Stack.stack[++top] = element;
      System.out.println(Stack.stack[top] + " pushed.");
    }
  }

  public static int pop()
  {
    if(isEmpty())
      return -999;
    else
      return Stack.stack[top--];
  }

  public static int peek()
  {
    if(isEmpty())
      return -999;
    else
      return Stack.stack[top];
  }

  public static void traverse()
  {
    if(isEmpty())
      System.out.println("Stack is empty.");
    else
    {
      System.out.println("Elements of stack top to bottom:");
      for(int stack_index = top; stack_index > -1; stack_index--)
        System.out.println(Stack.stack[stack_index]);
    }
  }
}
